src/components/Table/utils/RowState.ts

```diff
@@ -44,6 +44,13 @@ export class RowState {
     this.selected = !!row.initSelected;
     this.collapsed = states.storage.wasCollapsed(row.id) ?? !!row.initCollapsed;
     makeAutoObservable(this, { row: observable.ref }, { name: `RowState@${row.id}` });
+    const onSelect = row.onSelect;
```

Comment:
What are your thoughts on having a top level GridTableProps.onSelectionChange callback vs a callback per row?

Reply:
Hm, yeah, I think I like that--it keeps "the row is the data" better separated from "the behavior/logic that reacts to the row".

I guess a disclaimer is that we already have a fair amount of "select behavior" on the row, i.e. `initSelected`, `selectable`, and `inferSelectedState`...

Hm. Can't decide... Wdyt?

Reply:
We could probably have both... My thought with the invoicing v2 code was that the "on select" logic would be most clear right now to the code where they're creating the row from the form-state.

Here's the current `useEffect`:

https://github.com/homebound-team/internal-frontend/blob/main/src/routes/projects/invoices/invoice-v2/invoice-steps/SelectFixedItems.tsx#L101

And here's where I thought I'd move the logic, same file, but in createRows:

https://github.com/homebound-team/internal-frontend/blob/main/src/routes/projects/invoices/invoice-v2/invoice-steps/SelectFixedItems.tsx#L262

(I'm already adding an `initSelected` that they had been doing via another `useEffect`.)

Reply:
I think having an onSelectionChange as a root prop would be more clear than doing the API + useComputed combination that we have been doing everywhere.

Reply:
Hm, don't disagree re root prop, but not following where the API+useComputed would be needed for the row.onSelect?

The onSelect would be passed the row/data, and so could mutate it.

I guess if, inside the onSelect callback, they wanted to make a GQL call? They'd have to pass the mutation into the createRows fn, but, dunno, would not need the API or useComputed afaiu.

Reply:
I went ahead and added a `GridTable.onSelect`; we can tweak/remove one/the other as we/see how they work.